File: 02desafios/desafio3.py
def main():
    # dic = {
    #     'a': (stack.append('A') if not stack or stack[-1] == 'A' else stack.pop()),
    #     'b': (stack.append('B') if not stack or stack[-1] == 'B' else stack.pop()),
    # }

    # flag = 0
    stack = []
    buf = input()
    for i in buf:
        # dic[i]
        if i == 'a':
            if not stack or stack[-1] == 'A':
                stack.append('A')
            elif stack[-1] == 'B':
                stack.pop()
        elif i == 'b':
            if not stack or stack[-1] == 'B':
                stack.append('B')
            elif stack[-1] == 'A':
                stack.pop()
        if not stack:
            print('nil')
        else:
            print(''.join(stack))
    if not stack:
        print('Aceito')
    else:
        print('Rejeito')
    while True:
        try:
            stack = []
            buf = input()
            print()
            for i in buf:
                if i == 'a':
                    if not stack or stack[-1] == 'A':
                        stack.append('A')
                    elif stack[-1] == 'B':
                        stack.pop()
                elif i == 'b':
                    if not stack or stack[-1] == 'B':
                        stack.append('B')
                    elif stack[-1] == 'A':
                        stack.pop()
                if not stack:
                    print('nil')
                else:
                    print(''.join(stack))
            if not stack:
                print('Aceito')
            else:
                print('Rejeito')
        except EOFError:
            
            break

main()

File: 02desafios/desafio3.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    string buf;
    cin >> buf;
    stack<char> st;

    for (char i : buf){
        if(i == 'a'){
            if(st.empty()){
                st.push('A');
            }else if (st.top() == 'A'){
                st.push('A');
            }else if (st.top() == 'B'){
                st.pop();
            }
        }else if(st.empty()){
            st.push('B');
        }else if (st.top() == 'B'){
            st.push('B');
        }else if (st.top() == 'A'){
            st.pop();
        }
        if (st.empty()){
            cout << 'nil';
        }else{
            for(char j : st){
                printf('%c', st[j])
            }
        }
    }


    return 0;
}